include/seastar/coroutine/generator.hh

```diff
@@ -131,17 +131,20 @@ class generator_buffered_promise;
 template<typename T, template <typename> class Container>
 struct yield_awaiter final {
     using promise_type = generator_buffered_promise<T, Container>;
-    seastar::future<> _future;
+    const bool _wait_for_free_space;
 
 public:
-    yield_awaiter(seastar::future<>&& f) noexcept
-        : _future{std::move(f)} {}
+    yield_awaiter(bool wait_for_free_space) noexcept
+        : _wait_for_free_space{wait_for_free_space} {}
 
     bool await_ready() noexcept {
-        return _future.available();
+        return _wait_for_free_space;
+    }
+
+    coroutine_handle<> await_suspend(coroutine_handle<promise_type>) noexcept {
+        return std::noop_coroutine();
```

Comment:
I don't understand this. Why not return `true`?  Why resume some do-nothing coroutine?

Reply:
(or void)

Reply:
i don't want to return to the caller coroutine.